dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Internals/IOutputWriter.cs

```diff
@@ -0,0 +1,24 @@
+﻿/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */
+
+using System;
+
+namespace Razorvine.Pickle
+{
+    internal interface IOutputWriter : IDisposable
+    {
+        int BytesWritten { get; }
+        byte[] Buffer { get; }
+
+        void WriteByte(byte value);
+        void WriteBytes(byte first, byte second);
+        void WriteBytes(byte first, byte second, byte third);
+
+        void Write(byte[] buffer, int offset, int count);
+
+        void WriteInt32LittleEndian(int value);
+        void WriteInt64BigEndian(long value);
```

Comment:
I don't understand the pickling format, but having only `WriteInt32LittleEndian` and `WriteInt64BigEndian` seems a bit odd.

Reply:
I don't know why, but this is how floats and doubles are stored by Pickler. I kept the existing behavior.

https://github.com/irmen/Pyrolite/blob/f4fa6fce56692895573dba77487b30474537c63c/dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Pickler.cs#L620

Reply:
Yeah the pickle format is a bit strange at times. It goes back a long time so maybe that explains some of its odd choices. If you're interested I think the most accessible form of original  Python "documentation" or example code that describes its inner workings is https://github.com/python/cpython/blob/3.7/Lib/pickletools.py  (or ofcourse the pickle.py module itself for the actual serializer/deserializer).